deid/dicom/fields.py

```diff
@@ -250,8 +250,9 @@ def get_fields(dicom, skip=None, expand_sequences=True, seen=None):
     if not isinstance(skip, list):
         skip = [skip]
 
-    # Retrieve both dicom and file meta fields
-    datasets = [dicom, dicom.file_meta]
+    # Retrieve both dicom and file meta fields if dicom came from a file
+    datasets = [d for d in [dicom, dicom.get('file_meta', None)] if d is not None]
```

Comment:
a get defaults to None when it isn't there so you don't need to explicitly define it.
```suggestion
    datasets = [d for d in [dicom, dicom.get('file_meta')] if d is not None]
```
I think you can also just do (but I'd test first)

```suggestion
    datasets = [d for d in [dicom, dicom.get('file_meta')] if d]
```